syft/pkg/cataloger/deb/parse_dpkg_status_test.go

```diff
@@ -242,3 +259,135 @@ func TestSourceVersionExtract(t *testing.T) {
 		})
 	}
 }
+
+func assertAs(expected error) assert.ErrorAssertionFunc {
+	return func(t assert.TestingT, err error, i ...interface{}) bool {
+		return assert.ErrorAs(t, err, &expected)
+	}
+}
+
+func Test_parseDpkgStatus(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   string
+		want    []pkg.Package
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name:    "no more packages",
+			input:   `Package: apt`,
+			wantErr: assert.NoError,
+		},
+		{
+			name: "duplicated key",
+			input: `Package: apt
+Package: apt-get
+
+`,
+			wantErr: assertAs(errors.New("duplicate key discovered: Package")),
+		},
+		{
+			name: "no match for continuation",
+			input: `  Package: apt
+
+`,
+			wantErr: assertAs(errors.New("no match for continuation: line: '  Package: apt'")),
+		},
+		{
+			name: "find keys",
+			input: `Package: apt
+Status: install ok installed
+Installed-Size: 10kib
+
+`,
+			want: []pkg.Package{
+				{
+					Name:         "apt",
+					Type:         "deb",
+					MetadataType: "DpkgMetadata",
+					Metadata: pkg.DpkgMetadata{
+						Package:       "apt",
+						InstalledSize: 10240,
+						Files:         []pkg.DpkgFileRecord{},
+					},
+				},
+			},
+			wantErr: assert.NoError,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := bufio.NewReader(strings.NewReader(tt.input))
+			got, err := parseDpkgStatus(r)
+			tt.wantErr(t, err, fmt.Sprintf("parseDpkgStatus"))
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_handleNewKeyValue(t *testing.T) {
+	tests := []struct {
+		name    string
+		line    string
+		wantKey string
+		wantVal interface{}
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name:    "cannot parse field",
+			line:    "blabla",
+			wantErr: assertAs(errors.New("cannot parse field from line: 'blabla'")),
+		},
+		{
+			name:    "parse field",
+			line:    "key: val",
+			wantKey: "key",
+			wantVal: "val",
+			wantErr: assert.NoError,
+		},
+		{
+			name:    "parse installed size",
+			line:    "InstalledSize: 128",
+			wantKey: "InstalledSize",
+			wantVal: 128,
+			wantErr: assert.NoError,
+		},
+		{
+			name:    "parse installed kib size",
+			line:    "InstalledSize: 1kib",
+			wantKey: "InstalledSize",
+			wantVal: 1024,
+			wantErr: assert.NoError,
+		},
+		{
+			name:    "parse installed kb size",
+			line:    "InstalledSize: 1kb",
+			wantKey: "InstalledSize",
+			wantVal: 1000,
+			wantErr: assert.NoError,
+		},
+		{
+			name:    "parse installed-size mb",
+			line:    "Installed-Size: 1 mb",
+			wantKey: "InstalledSize",
+			wantVal: 1000000,
+			wantErr: assert.NoError,
+		},
+		{
+			name:    "fail parsing installed-size",
+			line:    "Installed-Size: 1bla",
+			wantKey: "",
+			wantErr: assertAs(fmt.Errorf("unhandled size name: %s", "bla")),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotKey, gotVal, err := handleNewKeyValue(tt.line)
+			tt.wantErr(t, err, fmt.Sprintf("handleNewKeyValue(%v)", tt.line))
```

Comment:
I would probably have `wantErr string` and then something like:
```
if tt.wantErr != "" {
  assertAs(fmt.Error(tt.wantErr))(t, err)
} else {
  assert.NoError(t, err)
}
```
... so each test case is made simpler

Reply:
Yes, I like the test running step be as simple as possible, where each test case has all the information for validation, so each case is just configuration.